### PR TITLE
Add raycast-ikea-idasen

### PR DESCRIPTION
**Pull request by akucharczyk**
## Description

Control your Idasen desk with raycast

## Screencast
Screenshot 2022-06-11 at 15 51 39
Screenshot 2022-06-11 at 15 51 30

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/raycast-ikea-idasen/package.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "raycast-ikea-idasen",
```
Comment (pernielsentikaer) [inline extensions/raycast-ikea-idasen/package.json:3]:
```suggestion
  "name": "ikea-idasen",
```
Rename this line and the extension folder

File: extensions/raycast-ikea-idasen/package.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  "title": "raycast-ikea-idasen",
```
Comment (pernielsentikaer) [inline extensions/raycast-ikea-idasen/package.json:4]:
```suggestion
  "title": "Ikea Idasen",
```

File: extensions/raycast-ikea-idasen/package.json [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  "description": "React component for IKEA IDASEN",
```
Comment (pernielsentikaer) [inline extensions/raycast-ikea-idasen/package.json:5]:
```suggestion
  "description": "Let's you control your IKEA IDASEN from within Raycast",
```

File: extensions/raycast-ikea-idasen/package.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  "author": "antoni",
+  "categories": [
+    "Productivity",
+    "Other"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Ikea Idasen",
```
Comment (pernielsentikaer) [inline extensions/raycast-ikea-idasen/package.json:16]:
```suggestion
      "title": "Table Control",
      "subtitle": "Ikea Idasen",
```

File: extensions/raycast-ikea-idasen/package.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      "description": "Set your Ikea Idasen to stand up position",
+      "keywords": [
+        "stand",
+        "up"
+      ],
+      "mode": "no-view"
+    },
+    {
+      "name": "sit-down",
+      "title": "Sit down",
```
Comment (pernielsentikaer) [inline extensions/raycast-ikea-idasen/package.json:33]:
```suggestion
      "title": "Sit Down",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @akucharczyk 👋 

Thanks for your contribution 💪 

I left some feedback and comments for you to look into:

- It seems like the installation check is not working, I can get into every command without seeing any errors. Would be nice to have an error toast (for view commands) and HUD for no-view
- 
I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @akucharczyk 👋 

Thanks for the corrections 🔥 

Since the extension gets unmounted when you close Raycast, then will the List.Item with `30 minutes` not work, and then there is no really need to have that command, I suggest removing it and just having the two other commands, doesn't that make sense? 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/ikea-idasen/package.json [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Table Control",
+      "subtitle": "Ikea Idasen",
+      "description": "Control your Ikea Idasen within raycast",
+      "mode": "view"
+    },
```
Comment (pernielsentikaer) [inline extensions/ikea-idasen/package.json:20]: Does it make sense to keep this now that they both have their own command?

Comment (akucharczyk) [reply extensions/ikea-idasen/package.json:20]: There is another command, which only will be shown when you not have installed the app, which you can download instantly. Later I also plan to add an option to add a custom high.

File: extensions/ikea-idasen/src/index.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  }, []);
+
+  return (
+    <List navigationTitle="Ikea Idasen">
+      <List.Item
+        icon={StandUpIcon}
+        title="Stand Up"
+        actions={
+          <ActionPanel>
+            <Action title="StandUp" onAction={standUp} />
```
Comment (pernielsentikaer) [inline extensions/ikea-idasen/src/index.tsx:25]:
```suggestion
            <Action title="Stand Up" onAction={standUp} />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @akucharczyk 👋 

Thanks for the update 💪 

The icons in the list doesn't look good in light mode, maybe you can use `ChevronDown` and `ChevronUp` for now? If not you have to specify another icon for light mode like:

`{ source: { light: "icon-light.png", dark: "icon-dark.png" } }`

CleanShot 2022-06-21 at 11 56 11@2x

File: extensions/ikea-idasen/package.json (hunk 6 above)
Comment (akucharczyk) [reply extensions/ikea-idasen/package.json:20]: I added the option to set a custom hight

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @akucharczyk 👋 

Thanks for the update, the new icons are much better 💯 

I'm however not sure about using the search bar for setting the custom height, maybe instead you could push to a new view and use a Form to add new custom heights - then it would be possible to add/edit/delete items in the list.

Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions regarding the above, I can help you out 😄

**Comment by akucharczyk** [comment]
> Hi @akucharczyk 👋
> 
> Thanks for the update, the new icons are much better 💯
> 
> I'm however not sure about using the search bar for setting the custom height, maybe instead you could push to a new view and use a Form to add new custom heights - then it would be possible to add/edit/delete items in the list.
> 
> Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions regarding the above, I can help you out 😄

I think this way is it much smoother than a form, because now you have a pretty good workflow. Sadly there is currently a bug from the plugin for the current version so I need to wait until they accept my mr and release a new version.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
I can see why you want to do that, but it's not a job for the search bar, it's not meant to do actions.

What you could do is to push to a new view where you can add a name and a title and then you can add custom sizes to the list, let me know if you need help

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by pernielsentikaer** [comment]
Are you still around @akucharczyk

**Comment by akucharczyk** [comment]
> 

Yes, but the application this shortcut work for has an bug. I already have an open pull-request for it, but waiting for the fix.
https://github.com/DWilliames/idasen-controller-mac/pull/17

**Comment by pernielsentikaer** [comment]
Great, converting to draft for now 😄 Thanks!

**Pull request closed without merging**